media-source-respec.html

```diff
@@ -2706,6 +2714,317 @@ <h3>Event Summary</h3>
+
+        dictionary BufferedChangeEventInit : EventInit {
+          TimeRanges addedRanges;
+          TimeRanges removedRanges;
+        };
+      </pre>
+      <h3>Attributes</h3>
+      <dl>
+        <dt>
+          <dfn>addedRanges</dfn>
+        </dt>
+        <dd>
+          The time ranges added during the last {{SourceBuffer/appendBuffer()}} operation.
+        </dd>
+        <dt>
+          <dfn>removedRanges</dfn>
+        </dt>
+        <dd>
+          The time ranges removed during the last {{SourceBuffer/remove()}} operation or if the
+          user agent evicted content in response to a memory pressure.
+        </dd>
+      </dl>
+    </section>
+
+    <section>
+      <h2><dfn>ManagedSourceBuffer</dfn> interface</h2>
+      <pre class="idl">
+        [Exposed=(Window,DedicatedWorker)]
+        interface ManagedSourceBuffer : SourceBuffer {
+          attribute EventHandler onbufferedchange;
+        };
+      </pre>
+      <h3>Attributes</h3>
+      <dl class="attributes" data-dfn-for="ManagedSourceBuffer">
+        <dt>
+          <dfn>onbufferedchange</dfn>
+        </dt>
+        <dd>
+          <p>The event handler for the {{bufferedchange}} event.</p>
+        </dd>
+      </dl>
+
+      <h3>Event Summary</h3>
+      <table class="old-table">
+        <thead>
+          <tr>
+            <th>Event name</th>
+            <th>Interface</th>
+            <th>Dispatched when...</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>
+              <dfn class="event">bufferedchange</dfn>
+            </td>
+            <td>{{Event}}</td>
+            <td>
+              The {{ManagedSourceBuffer}}'s buffered range changed following a
+              call to {{SourceBuffer/appendBuffer()}}, {{SourceBuffer/remove()}},
+              or as a consequence of the user agent running the
+              [=ManagedSourceBuffer/memory pressure=] algorithm.
+            </td>
+          </tr>
+        </tbody>
+      </table>
+
+      <h3>Algorithms</h3>
+      <h4>Buffered Change</h4>
+      <p>
+        The following steps are run at the completion of all operations to the {{ManagedSourceBuffer}}
+        |buffer:ManagedSourceBuffer| that would cause a |buffer|'s {{SourceBuffer/buffered}} to change.
+        That is once {{SourceBuffer/appendBuffer()}}, {{SourceBuffer/remove()}} or
+        [=ManagedSourceBuffer/memory pressure=] algorithm have completed.
+      </p>
+      <ol>
+        <li>
+          Let |previous buffered ranges:normalized TimeRanges| equal the
+          {{SourceBuffer/buffered}} attribute before the changes occurred.
+        </li>
+        <li>
+          Let |new buffered ranges:normalized TimeRanges| equal the new
+          {{SourceBuffer/buffered}} {{TimeRanges}}.
+        </li>
+        <li>
+          Let |added:normalized TimeRanges| equal the |previous buffered
+          ranges:normalized TimeRanges| subtracted from |new buffered
+          ranges:normalized TimeRanges|.
```

Comment:
Do we have `TimeRange` arithmetic defined somewhere?

Reply:
Just what Firefox’s TimeInterval arithmetic is doing :)